src/redprl/signature.sml

```diff
@@ -241,14 +241,18 @@ struct
     structure CJ = RedPrlCategoricalJudgment and Sort = RedPrlOpData and Hyps = RedPrlSequentData.Hyps
 
     fun elabAst (metactx, env) ast : abt =
-      let
-        val abt = AstToAbt.convertOpen (metactx, metactxToNameEnv metactx) (env, env) (ast, Sort.EXP)
-      in
-        abt
-      end
+      AstToAbt.convertOpen (metactx, metactxToNameEnv metactx) (env, env) (ast, Sort.EXP)
+
+    (* these should be included in the AstToAbt module *)
+    fun elabSym env (name : string) : Sym.t = NameEnv.lookup env name handle _ => Sym.named name
+
+    (* these should be included in the AstToAbt module *)
+    fun elabParam env = RedPrlParameterTerm.map (elabSym env)
+
+    fun elabLevel env = Option.map (RedPrlLevel.out o elabParam env o RedPrlAstLevel.into)
```

Comment:
Not completely sure about `elabLevel` and `elabParam`... `symctx` and `varctx` are not used at all.